Build fails for Oracle Linux 8 because of CMake target `test_constraint`

```
 [ 89%] Linking CXX executable test_constraint
/opt/rh/gcc-toolset-9/root/usr/bin/ld: CMakeFiles/test_constraint.dir/test_constraint.cpp.o: in function `BindingConstraintTests_FOO::BC_load_range_type::test_method()':
test_constraint.cpp:(.text+0x5a8): warning: the use of `tmpnam' is dangerous, better use `mkstemp'
/opt/rh/gcc-toolset-9/root/usr/bin/ld: /lib/../lib64/crt1.o: in function `_start':
(.text+0x24): undefined reference to `main'
collect2: error: ld returned 1 exit status
gmake[2]: *** [tests/src/libs/antares/study/constraint/CMakeFiles/test_constraint.dir/build.make:102: tests/src/libs/antares/study/constraint/test_constraint] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:9589: tests/src/libs/antares/study/constraint/CMakeFiles/test_constraint.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 89%] Linking CXX executable test-folder-output
[ 89%] Built target test-folder-output
gmake: *** [Makefile:166: all] Error 2
```

See [here](https://github.com/AntaresSimulatorTeam/Antares_Simulator/actions/runs/5488106623/jobs/10000481204).

--- a/.github/workflows/oracle8.yml
+++ b/.github/workflows/oracle8.yml
@@ -3,6 +3,15 @@ name: Oracle 8 CI (push and/or release)
 on:
   release:
     types: [created]
+  push:
+    branches:
+      - develop
+      - feature/*
+      - features/*
+      - fix/*
+      - issue-*
+      - release/*
+      - doc/*
 
 env:
   GITHUB_TOKEN: ${{ github.token }}

--- a/docker/oraclelinux
+++ b/docker/oraclelinux
@@ -1,0 +1,12 @@
+FROM oraclelinux:8
+
+# update repo
+RUN dnf update -y
+
+# install python & modules
+RUN dnf install -y python3 python3-pip
+RUN pip3 install --user pandas numpy pytest
+
+# Install build tools
+RUN dnf install -y epel-release git cmake wget rpm-build
+RUN dnf install -y unzip libuuid-devel boost-test boost-devel gcc-toolset-9-toolchain

--- a/src/tests/src/libs/antares/study/constraint/test_constraint.cpp
+++ b/src/tests/src/libs/antares/study/constraint/test_constraint.cpp
@@ -1,20 +1,21 @@
 //
 // Created by marechaljas on 13/03/23.
 //
-#include "antares/study/fwd.h"
-#define BOOST_TEST_MODULE binding_constraints_FOO
 
-#include <boost/test/unit_test.hpp>
+#define WIN32_LEAN_AND_MEAN
+#define BOOST_TEST_MODULE binding_constraints
+
+#include <boost/test/included/unit_test.hpp>
 #include <fstream>
 #include "antares/study/constraint.h"
 #include "antares/study/area/area.h"
-#include "antares/study/study.h"
+#include "antares/study.h"
 #include <filesystem>
 
 using namespace Antares::Data;
 namespace fs = std::filesystem;
 
-BOOST_AUTO_TEST_SUITE(BindingConstraintTests_FOO)
+BOOST_AUTO_TEST_SUITE(BindingConstraintTests)
 
 BOOST_AUTO_TEST_CASE( load_basic_attributes ) {
     auto study = std::make_shared<Study>();

--- a/src/tests/src/solver/simulation/test-store-timeseries-number.cpp
+++ b/src/tests/src/solver/simulation/test-store-timeseries-number.cpp
@@ -3,7 +3,7 @@
 //
 #define BOOST_TEST_MODULE store-timeseries-number
 #define WIN32_LEAN_AND_MEAN
-#include <boost/test/unit_test.hpp>
+#include <boost/test/included/unit_test.hpp>
 #include <filesystem>
 #include <fstream>
 #include "timeseries-numbers.h"

